src/ert/_c_wrappers/enkf/enkf_obs.py

```diff
@@ -26,18 +42,57 @@
 
 DEFAULT_TIME_DELTA = timedelta(seconds=30)
 
+USE_NEW_PARSER_BY_DEFAULT = True
 
-class ObservationConfigError(ConfigValidationError):
-    @classmethod
-    def get_value_error_message(cls, info: ErrorInfo) -> str:
-        return (
-            (
-                f"Parsing observations config file `{info.filename}` "
-                f"resulted in the following errors: {info.message}"
-            )
-            if info.filename is not None
-            else info.message
-        )
+if (
+    "USE_OLD_ERT_OBS_PARSER" in os.environ
+    and os.environ["USE_OLD_ERT_OBS_PARSER"] == "YES"
+):
+    USE_NEW_PARSER_BY_DEFAULT = False
+
+
+class _DictConfInstance:
+    def __init__(self, name: str, values: Mapping[str, Any]):
+        self.name = name
+        self.values = values
+
+    def __contains__(self, name: str) -> bool:
+        return name in self.values
+
+    def __getitem__(self, name: str) -> Any:
+        return self.values[name]
+
+
+class _HistoryConfInstance(_DictConfInstance):
+    def get_sub_instances(self, name: Literal["SEGMENT"]):
+        segments = self.values[name]
+        return [_DictConfInstance(segment[0], segment[1]) for segment in segments]
+
+
+class _NewParserAdapter:
+    def __init__(self, conf_content: ConfContent):
+        self.conf_content = conf_content
+
+    def get_sub_instances(self, name: str):
```

Comment:
Why not just break these down on initialization like 
```python 
self.history_observations =[]
self.general_observations =[]
self.summary_observations =[]

for _type, name, values, _ in self.conf_content: # ', _' required if there are more elements in the sub lists of self.conf_content
  if _type == ObservationType.HISTORY:
     self.history_observations.append(_HistoryConfInstance(name, values))
  if _type == ObservationType.GENERAL:
     self.general_observations.append(_DictConfInstance(name, values)) 
  if _type == ObservationType.SUMMARY:
     self.summary_observations.append(_DictConfInstance(name, values)) 
```

Ignore the above suggestion I realized `get_sub_instances` is required for compatibility reasons with other parts

Reply:
I think we can do something even simpler once the old parser is gone, and not sure if I can get the old parsers output to be iterated over like you suggested.